File: springboot-export-pdf/src/main/java/com/knf/dev/controller/EmployeeController.java
package com.knf.dev.controller;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.InputStreamResource;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import com.knf.dev.model.Employee;
import com.knf.dev.repository.EmployeeRepository;
import com.knf.dev.util.PDFGenerator;

@RestController
public class EmployeeController {

	@Autowired
	EmployeeRepository employeeRepository;

	@GetMapping(value = "/export", produces = MediaType.APPLICATION_PDF_VALUE)
	public ResponseEntity<InputStreamResource> employeeReport() throws IOException {

		List<Employee> employees = (List<Employee>) employeeRepository.findAll();

		ByteArrayInputStream bis = PDFGenerator.employeePDFReport(employees);

		HttpHeaders headers = new HttpHeaders();
		headers.add("Content-Disposition", "inline; filename=employees.pdf");

		return ResponseEntity.ok().headers(headers).contentType(MediaType.APPLICATION_PDF)
				.body(new InputStreamResource(bis));
	}
}

File: kotlin-spring-main/KotlinSpringExportPDF/src/main/kotlin/com/knf/dev/KotlinSpringExportPDF/controller/EmployeeController.kt
package com.knf.dev.KotlinSpringExportPDF.controller

import com.knf.dev.KotlinSpringExportPDF.model.Employee
import com.knf.dev.KotlinSpringExportPDF.repository.EmployeeRepository
import com.knf.dev.KotlinSpringExportPDF.util.PDFGenerator.employeePDFReport
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.io.InputStreamResource
import org.springframework.http.HttpHeaders
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.io.IOException


@RestController
@RequestMapping("/api/pdf")
class EmployeeController {
    @Autowired
    var employeeRepository: EmployeeRepository? = null

    @GetMapping(value = ["/employees"], produces = [MediaType.APPLICATION_PDF_VALUE])
    @Throws(
        IOException::class
    )
    fun employeeReport(): ResponseEntity<InputStreamResource> {
        val employees = employeeRepository!!.findAll() as List<Employee?>
        val bis = employeePDFReport(employees as List<Employee>)
        val headers = HttpHeaders()
        headers.add("Content-Disposition", "inline; filename=employees.pdf")
        return ResponseEntity.ok().headers(headers).contentType(MediaType.APPLICATION_PDF)
            .body(InputStreamResource(bis))
    }
}